### DISCUSSION TITLE
Rikaichamp in google docs

### DISCUSSION THREAD

**Discussion by PjPajkis** [birchill/10ten-ja-reader#780] [Q&A] [answered]
Hello,
firstly, thank you so much for making and updating rikaichamp, it helped me a lot, and I mean really a lot!

Lately I found out, that rikaichamp does not work in one google doc. Then I found out that in some docs it is working and in some it is not. I tried to make new google doc and it rikaichamp was not showing anything. If I make a copy of google doc, where it is working, then in the copy it does not. 

Is there any way how to solve this problem or does google do this intentionally?

**Comment by birtles** [comment, marked as answer]
Hi! Thank you for your kind words!

Yes, I'm aware of this issue and it's on my list of things to fix in the next week or two.

The problem is that Google is gradually rolling out a new "canvas mode" which doesn't expose the text of the documents to extensions like 10ten (Rikaichamp). It appears to be somewhat random which documents get the new "canvas mode" and which ones get the old behavior. That's why sometimes opening the document in a different tab might cause it to work.

It appears to be possible to force Google docs to use the old mode so I plan to add that functionality to 10ten in the next few week or two.

Thank you very much for your patience!

**Comment by birtles** [comment]
I've just released v1.4.3 which makes Google Docs work correctly for now. Apparently Google Docs will prevent this workaround from working later this year, however, so it may break again in the coming months, unfortunately.

**Reply by PjPajkis** [reply]
Sorry for replying super late and thank you very much!

**Reply by birtles** [reply]
No problem. I've since released v1.5.0 which works with the new Google Docs without relying on a workaround.
